marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -257,10 +258,7 @@ void processOutputBatch(InputStream[] output, Consumer<InputStream> outputConsum
             if (allowsEndpointState() && output.length > 0) {
                 setEndpointState(output[0]);
             }
```

Comment:
Does the code that caches the endpoint state (lines 258 through 259 above) execute in the next() case or only in the output consumer case?